### PR TITLE
LPS-187665 Do not allow to delete Custom Object Fields after publish a Modifiable System Object Definition

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://liferay.atlassian.net/browse/LPS-187665

### CODE REVIEW

**Comment by mateussgn** [comment]
ci:reevaluate:1347401_4348

**Review by p-albuquerque — APPROVED** [review]

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsFieldsDisplayContext.java [hunk 1]
```diff
@@ -122,6 +122,18 @@ public List<FDSActionDropdownItem> getFDSActionDropdownItems()
...
+
+		if (objectDefinition.isApproved() && objectDefinition.isModifiable() &&
+			objectDefinition.isSystem()) {
+
+			return Arrays.asList(
+				new FDSActionDropdownItem(
+					getEditObjectFieldURL(), "view", "view",
+					LanguageUtil.get(objectRequestHelper.getRequest(), "view"),
+					"get", null, "sidePanel"));
+		}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsFieldsDisplayContext.java:135]: @mateussgn you don't need to this,

you just need to make sure the delete action will be retrieved as null [here](https://github.com/liferay/liferay-portal/blob/357f569f2ac814e50cf64a624b830a840748e576/modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFieldResourceImpl.java#L332-L343),

Should we adapt the method [objectField.isDeletionAllowed](https://github.com/liferay/liferay-portal/blob/357f569f2ac814e50cf64a624b830a840748e576/modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectFieldResourceImpl.java#L334) as well?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java [hunk 2]
```diff
@@ -982,6 +983,13 @@ else if (!deleteRelationshipObjectField) {
 
 				throw new RequiredObjectFieldException();
 			}
+
+			if (objectDefinition.isApproved() &&
+				objectDefinition.isModifiable() &&
+				objectDefinition.isSystem()) {
+
+				throw new ApprovedSystemModifiableObjectDefinitionException();
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java:991]: @mateussgn I'm not sure about the name we should give to this exception,

use UnsupportedOperationException, I saw some examples in objects and makes sense to me for while

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsFieldsDisplayContext.java (hunk 1 above)
Comment (mateussgn) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/ObjectDefinitionsFieldsDisplayContext.java:135]: Great catch! Thank you!

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java (hunk 2 above)
Comment (mateussgn) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java:991]: Ok! I'll change it.

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
Hey, @MarinhoFeliphe, I made the changes, can you review it again?

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

**Comment by mateussgn** [comment]
ci:forward:force

**Pull request closed without merging**